enumerable.rb

```diff
@@ -0,0 +1,72 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    temp = self
+    i = 0
+    while i < temp.length
+      yield(temp[i])
+      i += 1
+    end
+  end
+
+  def my_each_with_index
+    temp = self
+    i = 0
+    while i < temp.length
+      yield(temp[i], i)
+      i += 1
+    end
+  end
+
+  def my_select
+    temp = []
+    my_each { |e| temp << e if yield(e) }
+    temp
+  end
+
+  def my_all?
+    test = true
+    my_each { |d| break unless test == yield(d) }
+    test
+  end
+
+  def my_any?
+    test = false
+    my_each { |u| break if test == yield(u) }
+    test
+  end
+
```

Comment:
In `my_any?`
- enumerables my_any? returns true if the block ever returns a value other than false or nil
- enumerables my_any? when no block or argument is given returns true if at least one of the collection is not false or nil
- enumerables my_any? when a class is passed as an argument returns true if at least one of the collection is a member of such class
- enumerables my_any? when a Regex is passed as an argument returns false if none of the collection matches the Regex
- enumerables my_any? when a pattern other than Regex or a Class is given returns false if none of the collection matches the pattern